CRM/Member/BAO/Membership.php

```diff
@@ -1915,6 +1915,9 @@ public static function createRelatedMemberships(&$params, &$dao, $reset = FALSE)
         unset($params['lineItems']);
         unset($params['line_item']);
 
+        // CRM-20966: Do not create membership_payment record for inherited membership.
+        unset($params['relate_contribution_id']);
+
```

Comment:
You probably already looked at this - but can we prevent it from getting "set" in the first place?